src/commands/createHttpFunction.ts

```diff
@@ -61,7 +61,18 @@ function generateSuffixedName(preferredName: string, i: number): string {
 }
 
 async function isNameAvailable(folderPath: string, newName: string): Promise<boolean> {
-    return !(await fse.pathExists(path.join(folderPath, newName)));
+    // if the file or folder doesn't exist, it will throw an error so we have to wrap these in try/catches
+    try {
+        await fse.access(path.join(folderPath, newName));
+    } catch (error) {
```

Comment:
>  I felt like it made the logic a little simpler rather than doing pathExists, returning it if it's true, if it's false, go into another statement to check if the file exists, then try/catch that and return the result.

IMO, the problem with this is that we don't know why the error occurred. _One_ of the possible errors is that it doesn't exist, but there's more possible errors. It's safer to keep using pathExists